From: Kiran Kumar Modukuri <kiran.modukuri@gmail.com>
Subject: cachefiles: Wait rather than BUG'ing on "Unexpected object collision"
Git-commit: c2412ac45a8f8f1cd582723c1a139608694d410d
Patch-mainline: v4.18-rc7
References: Git-fixes bsc#1109749

If we meet a conflicting object that is marked FSCACHE_OBJECT_IS_LIVE in
the active object tree, we have been emitting a BUG after logging
information about it and the new object.

Instead, we should wait for the CACHEFILES_OBJECT_ACTIVE flag to be cleared
on the old object (or return an error).  The ACTIVE flag should be cleared
after it has been removed from the active object tree.  A timeout of 60s is
used in the wait, so we shouldn't be able to get stuck there.

Fixes: 9ae326a69004 ("CacheFiles: A cache that backs onto a mounted filesystem")
Signed-off-by: Kiran Kumar Modukuri <kiran.modukuri@gmail.com>
Signed-off-by: David Howells <dhowells@redhat.com>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 fs/cachefiles/namei.c | 1 -
 1 file changed, 1 deletion(-)

--- a/fs/cachefiles/namei.c
+++ b/fs/cachefiles/namei.c
@@ -199,7 +199,6 @@ wait_for_old_object:
 		printk(KERN_ERR "CacheFiles: Error:"
 		       " Unexpected object collision\n");
 		cachefiles_printk_object(object, xobject);
-		BUG();
 	}
 	atomic_inc(&xobject->usage);
 	write_unlock(&cache->active_lock);

